internal/creds.go

```diff
@@ -90,9 +90,8 @@ func credentialsFromJSON(ctx context.Context, data []byte, endpoint string, scop
 	return cred, err
 }
 
-func selfSignedJWTTokenSource(data []byte, endpoint string, audiences []string) (oauth2.TokenSource, error) {
-	// Use the API endpoint as the default audience
-	audience := endpoint
+func selfSignedJWTTokenSource(data []byte, defaultAudience string, audiences []string) (oauth2.TokenSource, error) {
+	audience := defaultAudience
```

Comment:
Doesn't this imply that `audiences` must always be empty?

Reply:
L100 below will override the audience if the user provided one with a client option.

Reply:
Ahh yes, I confused `audience` with `audiences` and thought `:=` was an array push.